src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -78,6 +81,96 @@ public AWSResourceQueryer(IAWSClientFactory awsClientFactory)
             _awsClientFactory = awsClientFactory;
         }
 
+        public async Task<List<int>> ListOfAvailableInstanceCPUCores(bool freeTierEligible, string architecture)
+        {
+            var ec2Client = _awsClientFactory.GetAWSClient<IAmazonEC2>();
+            var instanceTypes = new List<InstanceTypeInfo>();
+            var listInstanceTypesPaginator = ec2Client.Paginators.DescribeInstanceTypes(new DescribeInstanceTypesRequest
+            {
+                Filters = new List<Filter> {
+                    new Filter {
+                        Name = "free-tier-eligible",
+                        Values = new List<string>{ freeTierEligible.ToString().ToLower() }
+                    },
+                    new Filter {
+                        Name = "processor-info.supported-architecture",
+                        Values = new List<string>{ architecture }
+                    }
+                }
+            });
+
+            await foreach (var response in listInstanceTypesPaginator.Responses)
+            {
+                instanceTypes.AddRange(response.InstanceTypes);
+            }
+
+            return instanceTypes.Select(x => x.VCpuInfo.DefaultCores).Distinct().OrderBy(x => x).ToList();
+        }
+
+        public async Task<List<double>> ListOfAvailableInstanceMemory(bool freeTierEligible, string architecture, string cpuCores)
+        {
+            var ec2Client = _awsClientFactory.GetAWSClient<IAmazonEC2>();
+            var instanceTypes = new List<InstanceTypeInfo>();
+            var listInstanceTypesPaginator = ec2Client.Paginators.DescribeInstanceTypes(new DescribeInstanceTypesRequest
+            {
+                Filters = new List<Filter> {
+                    new Filter {
+                        Name = "free-tier-eligible",
+                        Values = new List<string>{ freeTierEligible.ToString().ToLower() }
+                    },
+                    new Filter {
+                        Name = "processor-info.supported-architecture",
+                        Values = new List<string>{ architecture }
+                    },
+                    new Filter {
+                        Name = "vcpu-info.default-cores",
+                        Values = new List<string>{ cpuCores }
+                    }
+                }
+            });
+
+            await foreach (var response in listInstanceTypesPaginator.Responses)
+            {
+                instanceTypes.AddRange(response.InstanceTypes);
+            }
+
+            return instanceTypes.Select(x => (double) x.MemoryInfo.SizeInMiB).Distinct().OrderBy(x => x).ToList();
```

Comment:
Bases on the API ref, the `SizeInMiB` is an `int`. Why are we casting it into a double?

API ref - https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/EC2/TMemoryInfo.html

Reply:
In the API ref you linked, it says that it is an Int64 which is mapped to a `long`. I casted the `long` to `double` since we currently support the type `Double` in our recipes.